### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by M-AminAlizadeh**
The things that we did in this PR:

- On startup of the app load all data from files
- On exit from the app save all the data (books, people, and rentals) into JSON files

### CODE REVIEW

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Good descriptive README file 👍🏽

## Required Changes ♻️
_Check the comments under the review._

I am leaving the review feedback here because I couldn't find the file changes in the specific areas where things are not working as expected. Kindly work on the changes mentioned here and resubmit for review 👍🏽 

- [ ] When creating a person, the app keeps asking for the creation of a person repeatedly. It's expected that, after creating either a student or a teacher, the creation completes and then brings the initial options for other features. Kindly fix that so the option for creating a person does not keep looping infinitely 👍🏽

Screenshot 2023-10-11 at 8 06 35 PM

- [ ] I tried creating a book and the app crashed with the error shown in the screenshot below. Kindly work on that so a book can be created successfully 👍🏽

Screenshot 2023-10-11 at 8 05 09 PM

- [ ] I am unable to check if data is stored as expected because the app is not working as expected for data to be stored in separate JSON files as expected. Kindly fix the issues mentioned above so all features work fine and data is preserved as expected 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: rentals.json
Comment (shella12) [inline rentals.json:1]: - [ ] Great job on working on the changes requested by the previous reviewer. The data presevation feature ia working well for books and person however when we exit the app rental is not being saved and is causing an error. Please try to fix the error and see the screenshot for error details:

Screenshot 2023-10-12 at 1 38 25 PM

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @M-AminAlizadeh, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. You are almost there...



## Highlights

- Data is preserved in `json` format ✅ 
- Console app is working as expected ✅   

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rentals.json
Comment (M-AminAlizadeh) [reply rentals.json:1]: @shella12 Thank you for the review
Yes I missed this one

**Review by Munsa1 — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
App Works As Expected :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**